File: frontend/src/api/schemas/dtos/schema-draft-diff-dto.ts
/*
SPDX-License-Identifier: MIT
Copyright (c) 2025 Pablo Ulloa Santin
*/

import type { SchemaDraftChangeDto } from "./schema-draft-change-dto";

export type SchemaDraftDiffDto = {
  baseVersionId: string;
  currentVersionId: string;
  currentDocumentHash: string;
  hasConflicts: boolean;
  changes: SchemaDraftChangeDto[];
};

File: api/src/main/java/dev/ulloasp/mlsuite/schema/application/dto/SchemaDraftDiffDto.java
package dev.ulloasp.mlsuite.schema.application.dto;

import java.util.List;

public record SchemaDraftDiffDto(
        Long baseVersionId,
        Long currentVersionId,
        String currentDocumentHash,
        boolean hasConflicts,
        List<SchemaDraftChangeDto> changes) {
}
